homeassistant/components/enocean/switch.py

```diff
@@ -25,6 +27,26 @@
 )
 
 
+def generate_unique_id(dev_id: list[int], channel: int) -> str:
+    """Generate a valid unique id."""
+    return f"{combine_hex(dev_id)}{channel}"
+
+
+def _migrate_to_new_unique_id(hass: HomeAssistant, dev_id, channel) -> None:
+    """Migrate old unique ids to new unique ids."""
+    old_unique_id = f"{combine_hex(dev_id)}"
+
+    ent_reg = entity_registry.async_get(hass)
+    entity_id = ent_reg.async_get_entity_id(Platform.SWITCH, DOMAIN, old_unique_id)
+
+    if entity_id is not None:
+        new_unique_id = generate_unique_id(dev_id, channel)
+        try:
+            ent_reg.async_update_entity(entity_id, new_unique_id=new_unique_id)
+        except ValueError:
+            pass
+
+
 def setup_platform(
```

Comment:
The async api needs to run inside the event loop.
```suggestion
async def async_setup_platform(
```

Reply:
Done ✅